java/dagger/hilt/processor/internal/root/ir/ComponentTreeDepsIrCreator.kt

```diff
@@ -133,10 +133,10 @@ class ComponentTreeDepsIrCreator private constructor(
     val globalEntryPointsByComponent = aggregatedDeps
       .filter { it.test == null && it.module == null }
       .groupBy(keySelector = { it.test }, valueTransform = { it.fqName })
-    val result = mutableMapOf<ClassName, MutableSet<ClassName>>()
+    val result = mutableMapOf<ClassName, LinkedHashSet<ClassName>>()
     aggregatedRoots.forEach { aggregatedRoot ->
       if (!rootsUsingSharedComponent.contains(aggregatedRoot.root)) {
-        result.getOrPut(aggregatedRoot.root) { mutableSetOf() }.apply {
+        result.getOrPut(aggregatedRoot.root) { linkedSetOf() }.apply {
```

Comment:
I actually thought by default Kotlin uses `LinkedHashSet`:
```
@SinceKotlin("1.1")
@kotlin.internal.InlineOnly
public inline fun <T> mutableSetOf(): MutableSet<T> = LinkedHashSet()
```

Reply:
Darn, I'll have to keep looking then. I have some follow-up questions, but I'll ask on the bug.

Regardless, I'll submit this since it's better to be explicit when we rely on the order.